orion-server/src/main/java/com/pinterest/orion/core/automation/operator/kafka/BrokersetTopicOperator.java

```diff
@@ -166,6 +166,15 @@ public void operate(KafkaCluster cluster) throws Exception {
 
       if (idealPartitionCount > actualPartitionCount) {
         // expansion
+        // reject expasion if topic has compact in cleanup policy
+        if (topicAssignment.getConfig() != null &&
+                topicAssignment.getConfig().containsKey("cleanup.policy") &&
+                topicAssignment.getConfig().get("cleanup.policy").contains("compact")) {
```

Comment:
Looks OK to me, but a more bullet-proof check would be splitting the string by `,` and looking for `compact`.
That would guard against a possibility that another policy with `compact` in it could appear. :)
Not a big deal.